src/credentials/FabricTable.cpp

```diff
@@ -994,9 +1014,40 @@ CHIP_ERROR FabricTable::Init(const FabricTable::InitParams & initParams)
         TLV::ContiguousBufferTLVReader reader;
         reader.Init(buf, size);
 
-        ReturnErrorOnFailure(ReadFabricInfo(reader));
+        // TODO: A safer way would be to just clean-up the entire fabric table on this situation...
+        err = ReadFabricInfo(reader);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(FabricProvisioning, "Error loading fabric table: %" CHIP_ERROR_FORMAT ", we are in a bad state!",
+                         err.Format());
+        }
+
+        ReturnErrorOnFailure(err);
     }
 
+    CommitMarker commitMarker;
+    err = GetCommitMarker(commitMarker);
+    if (err == CHIP_NO_ERROR)
+    {
+        // Found a commit marker! We need to possibly delete a loaded fabric
+        ChipLogError(FabricProvisioning, "Found a FabricTable aborted commit for index 0x%x (isAddition: %d), removing!",
+                     static_cast<unsigned>(commitMarker.fabricIndex), static_cast<int>(commitMarker.isAddition));
+
+        mDeletedFabricIndexFromInit = commitMarker.fabricIndex;
+
+        // Can't do better on error. We just have to hope for the best.
+        (void) Delete(commitMarker.fabricIndex);
+    }
+    else if (err != CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND)
+    {
+        // Got an error, but somehow value is not missing altogether: inconsistent state but touch nothing.
+        ChipLogError(FabricProvisioning, "Error loading Table commit marker: %" CHIP_ERROR_FORMAT ", hope for the best!",
+                     err.Format());
+    }
+
+    // Always statelessly revert commit marker in this situation.
+    ClearCommitMarker();
```

Comment:
So if we reboot again after this `ClearCommitMarker` call but before whatever action the consumer is expected to take based on the commit marker... won't that action just get lost?

Reply:
I think the rest is best effort. If I put the ClearCommitMarker in some event handler, when is the clean-up done? Like, which event is the winner? There is no clear order of execution, and it's initiated via asynchronous events.

To fix this, we would need to refactor FabricTable to be the locus of all RemoveFabric side-effects, which is possible to a degree, but I'm concenred about the risk right now.

I think the way it's done now, it's "better than before, but not perfect"

Reply:
Fixed